x/auth/ante/basic.go

```diff
@@ -79,9 +80,29 @@ type ConsumeTxSizeGasDecorator struct {
 	ak AccountKeeper
 }
 
-func NewConsumeGasForTxSizeDecorator(ak AccountKeeper) ConsumeTxSizeGasDecorator {
-	return ConsumeTxSizeGasDecorator{
-		ak: ak,
+type ConsumeTxSizeGasDepDecorator struct {}
+
+func (d ConsumeTxSizeGasDepDecorator) AnteDeps(txDeps []sdkacltypes.AccessOperation, tx sdk.Tx, next sdk.AnteDepGenerator) (newTxDeps []sdkacltypes.AccessOperation, err error) {
+	sigTx, _ := tx.(authsigning.SigVerifiableTx)
+	deps := []sdkacltypes.AccessOperation{}
+	for _, signer := range sigTx.GetSigners() {
+		deps = append(deps,
+			// Consumes gas fee!
+			sdkacltypes.AccessOperation{
+				AccessType:         sdkacltypes.AccessType_WRITE,
+				ResourceType:       sdkacltypes.ResourceType_KV,
+				IdentifierTemplate:  signer.String(),
+			},
+		)
+
+	}
+	return next(append(txDeps, deps...), tx)
+}
+
+func NewConsumeGasForTxSizeDecorator(ak AccountKeeper) sdk.WrappedAnteDecorator {
```

Comment:
here we can instead just return `ConsumeTxSizeGasDecorator` and implement AnteDeps for that struct